### PR TITLE
port the module to gatling 3.0.3

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: src/main/scala-2.12/com/linagora/gatling/imap/action/ImapResponseHandler.scala [hunk 1]
```diff
@@ -25,8 +23,8 @@ class ImapResponseHandler(imapActionContext: ImapActionContext, requestName: Str
 
   override def receive: Receive = {
     case Response(responses) =>
-      val (checkSaveUpdate, error) = Check.check(responses, session, checks.toList)
-      error.fold(ok(session, start, checkSaveUpdate))(ko(session, start))
+      val (newSession, error) = Check.check(responses, session, checks.toList)
```
Comment (mbaechler) [inline src/main/scala-2.12/com/linagora/gatling/imap/action/ImapResponseHandler.scala]: could someone check this specific part, I'm not sure I understand it correctly

File: build.sbt [hunk 2]
```diff
@@ -2,15 +2,19 @@ name := "gatling-imap"
...
+val gatlingVersion = "3.0.3"
+
+scalacOptions in ThisBuild ++= Seq("-unchecked", "-deprecation", "-feature", "-language:postfixOps")
 
 enablePlugins(GatlingPlugin)
 
-libraryDependencies += "io.gatling" % "gatling-test-framework" % "2.2.2" exclude("io.gatling", "gatling-http")
-libraryDependencies += "io.gatling.highcharts" % "gatling-charts-highcharts" % "2.2.2" exclude("io.gatling", "gatling-http")
+libraryDependencies += "io.gatling" % "gatling-test-framework" % gatlingVersion exclude("io.gatling", "gatling-http")
+libraryDependencies += "io.gatling.highcharts" % "gatling-charts-highcharts" % gatlingVersion exclude("io.gatling", "gatling-http")
```
Comment (rouazana) [inline build.sbt]: did you try to remove the exclusion (could be a good indicator of next ticket easiness ;))?

File: src/main/scala-2.12/com/linagora/gatling/imap/protocol/ImapProtocol.scala [hunk 3]
```diff
@@ -2,33 +2,31 @@ package com.linagora.gatling.imap.protocol
...
+    override def newComponents(coreComponents: CoreComponents): ImapProtocol => ImapComponents = { protocol =>
+      val sessions: ActorRef = coreComponents.actorSystem.actorOf(ImapSessions.props(protocol), "imapsessions")
       ImapComponents(protocol, sessions)
     }
   }
 }
 
 case class ImapComponents(protocol: ImapProtocol, sessions: ActorRef) extends ProtocolComponents {
-  override def onStart: Option[(Session) => Session] = None
+  override def onStart: Session => Session = s => s
```
Comment (blackheaven) [inline src/main/scala-2.12/com/linagora/gatling/imap/protocol/ImapProtocol.scala]: Is there identity function in Scala?

Comment (remk) [reply src/main/scala-2.12/com/linagora/gatling/imap/protocol/ImapProtocol.scala]: I didn't know until now but you have one in scala.Predef.identity

File: src/main/scala-2.12/com/linagora/gatling/imap/action/ImapResponseHandler.scala (hunk 1 above)
Comment (remk) [reply src/main/scala-2.12/com/linagora/gatling/imap/action/ImapResponseHandler.scala]: seems good to me

File: src/main/scala-2.12/com/linagora/gatling/imap/protocol/ImapProtocol.scala (hunk 3 above)
Comment (blackheaven) [reply src/main/scala-2.12/com/linagora/gatling/imap/protocol/ImapProtocol.scala]: It's more expressive to use it, and avoid handcrafted expressions

File: build.sbt (hunk 2 above)
Comment (mbaechler) [reply build.sbt]: Yes, that's the following PR

**Pull request merged**